internal/urlparser_test.go

```diff
@@ -0,0 +1,139 @@
+package internal
+
+import (
+	"fmt"
+	"net/url"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// This test is more like documentation of the golang's net/url
```

Comment:
Shouldn't this test be removed when the code comments were in place?

Reply:
Yes, but these tests are similar but different. It tests the functionality of the parseEngineURL method and validates that the returned URL is correct using reflection.